src/provider/model/ClientRegistry.ts

```diff
@@ -56,28 +56,31 @@ export class ClientRegistry {
     }
 
     /**
-     * Decides whether an app is running at the moment.
+     * Decides whether an app is ready to receive `notification-action` events
      *
      * @param appUuid Uuid of the app.
      */
-    public isAppActive(appUuid: string): boolean {
-        return this._activeClients.some(client => client.uuid === appUuid);
+    public isAppActionReady(appUuid: string): boolean {
+        return this._actionReadyWindows.some(client => client.uuid === appUuid);
     }
 
-    public getAllAppWindows(uuid: string): Identity[] {
-        return this._activeClients.filter((client) => {
-            return client.uuid === uuid;
-        });
+    public getConnectedWindowsByApp(uuid: string): Identity[] {
```

Comment:
Should this an `APIHandler` method?

As in `getClientConnections`, and `getClientConnectionsByApp`?

Reply:
My instinct is no, I feel like let's not push very use-case specific calls into `APIHander`. But no strong objection if you're keen on this

Reply:
Fixed as discussed

✅